tests/Aspire.Hosting.Tests/Utils/TestDistributedApplicationBuilder.cs

```diff
@@ -109,6 +111,13 @@ void Configure(DistributedApplicationOptions applicationOptions, HostApplication
     {
         return _innerBuilder.CreateResourceBuilder(resource);
     }
+    public Task WaitForText(IResource resource, string text, CancellationToken cancellationToken = default)
```

Comment:
I wonder if this should look more like what it does in https://github.com/dotnet/aspire/pull/4444.

```C#
var resourceNotificationService = app.Services.GetRequiredService<ResourceNotificationService>();
await app.StartAsync();

await resourceNotificationService.WaitForResourceAsync("webfrontend", KnownResourceStates.Running).WaitAsync(TimeSpan.FromSeconds(30));
```

But here instead of waiting for `KnownResourceStates.Running`, we are waiting for a log message.

Reply:
It's a different kind of pattern since you can't directly await the result of this method since you have to call it before you build the app.